### PR TITLE
Modify person and add tutorial group

### PR DESCRIPTION
**Pull request by LowJiaHao99**

### CODE REVIEW

File: src/main/java/seedu/address/model/person/Student.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+    public GitHub getGitHub() {
+        return gitHub;
+    }
+
+    /**
+     * Returns an immutable tag set, which throws {@code UnsupportedOperationException}
+     * if modification is attempted.
+     */
+    public Set<TutorialGroup> getTags() {
```
Comment (jaysmyname) [inline src/main/java/seedu/address/model/person/Student.java:60]: change method name

File: src/main/java/seedu/address/model/person/Student.java [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    }
+
+    /**
+     * Returns true if both persons have the same name.
+     * This defines a weaker notion of equality between two persons.
+     */
+    public boolean isSamePerson(Student otherStudent) {
+        if (otherStudent == this) {
+            return true;
+        }
```
Comment (jaysmyname) [inline src/main/java/seedu/address/model/person/Student.java:71]: .

File: src/main/java/seedu/address/model/person/Student.java [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+        return otherStudent != null
+                && otherStudent.getName().equals(getName());
+    }
+
+    /**
+     * Returns true if both persons have the same identity and data fields.
+     * This defines a stronger notion of equality between two persons.
+     */
```
Comment (jaysmyname) [inline src/main/java/seedu/address/model/person/Student.java:80]: .

File: src/main/java/seedu/address/model/person/UniquePersonList.java [hunk 4]
```diff
@@ -13,34 +13,35 @@
...
+ * A student is considered unique by comparing using {@code Student#isSamePerson(Student)}.
+ * As such, adding and updating of persons uses Student#isSameStudent(Student) for equality
+ * so as to ensure that the student being added or updated is unique in terms of identity in the UniquePersonList.
+ * However, the removal of a student uses Student#equals(Object) so
+ * as to ensure that the student with exactly the same fields will be removed.
  *
  * Supports a minimal set of list operations.
  *
- * @see Person#isSamePerson(Person)
+ * @see Student#isSamePerson(Student)
```
Comment (jaysmyname) [inline src/main/java/seedu/address/model/person/UniquePersonList.java:24]: .

File: src/main/java/seedu/address/model/person/UniquePersonList.java [hunk 5]
```diff
@@ -13,34 +13,35 @@
...
  */
-public class UniquePersonList implements Iterable<Person> {
+public class UniquePersonList implements Iterable<Student> {
```
Comment (jaysmyname) [inline src/main/java/seedu/address/model/person/UniquePersonList.java:26]: .

File: src/main/java/seedu/address/model/person/UniquePersonList.java [hunk 6]
```diff
@@ -85,27 +86,27 @@ public void setPersons(UniquePersonList replacement) {
...
-     * {@code persons} must not contain duplicate persons.
+     * Replaces the contents of this list with {@code students}.
+     * {@code students} must not contain duplicate students.
      */
-    public void setPersons(List<Person> persons) {
-        requireAllNonNull(persons);
-        if (!personsAreUnique(persons)) {
+    public void setPersons(List<Student> students) {
+        requireAllNonNull(students);
+        if (!personsAreUnique(students)) {
```
Comment (jaysmyname) [inline src/main/java/seedu/address/model/person/UniquePersonList.java:94]: change method name

File: src/main/java/seedu/address/model/person/UniquePersonList.java [hunk 7]
```diff
@@ -122,12 +123,12 @@ public int hashCode() {
...
+     * Returns true if {@code students} contains only unique students.
      */
-    private boolean personsAreUnique(List<Person> persons) {
-        for (int i = 0; i < persons.size() - 1; i++) {
-            for (int j = i + 1; j < persons.size(); j++) {
-                if (persons.get(i).isSamePerson(persons.get(j))) {
+    private boolean personsAreUnique(List<Student> students) {
+        for (int i = 0; i < students.size() - 1; i++) {
+            for (int j = i + 1; j < students.size(); j++) {
+                if (students.get(i).isSamePerson(students.get(j))) {
```
Comment (jaysmyname) [inline src/main/java/seedu/address/model/person/UniquePersonList.java:131]: change method name

File: src/main/java/seedu/address/model/person/exceptions/PersonNotFoundException.java [hunk 8]
```diff
@@ -1,6 +1,6 @@
 package seedu.address.model.person.exceptions;
 
 /**
- * Signals that the operation is unable to find the specified person.
+ * Signals that the operation is unable to find the specified student.
  */
 public class PersonNotFoundException extends RuntimeException {}
```
Comment (jaysmyname) [inline src/main/java/seedu/address/model/person/exceptions/PersonNotFoundException.java:6]: .

File: src/main/java/seedu/address/model/tutorialgroup/TutorialGroup.java [hunk 9]
```diff
@@ -1,42 +1,43 @@
...
-     * Constructs a {@code Tag}.
+     * Constructs a {@code TutorialGroup}.
      *
      * @param tagName A valid tag name.
      */
-    public Tag(String tagName) {
+    public TutorialGroup(String tagName) {
         requireNonNull(tagName);
-        checkArgument(isValidTagName(tagName), MESSAGE_CONSTRAINTS);
+        checkArgument(isValidTutorialGroupName(tagName), MESSAGE_CONSTRAINTS);
```
Comment (jaysmyname) [inline src/main/java/seedu/address/model/tutorialgroup/TutorialGroup.java:25]: .

File: src/main/java/seedu/address/model/tutorialgroup/TutorialGroup.java [hunk 10]
```diff
@@ -1,42 +1,43 @@
...
         return test.matches(VALIDATION_REGEX);
     }
 
     @Override
     public boolean equals(Object other) {
         return other == this // short circuit if same object
-                || (other instanceof Tag // instanceof handles nulls
-                && tagName.equals(((Tag) other).tagName)); // state check
+                || (other instanceof TutorialGroup // instanceof handles nulls
+                && tagName.equals(((TutorialGroup) other).tagName)); // state check
```
Comment (jaysmyname) [inline src/main/java/seedu/address/model/tutorialgroup/TutorialGroup.java:40]: .

File: src/main/java/seedu/address/storage/JsonSerializableAddressBook.java [hunk 11]
```diff
@@ -11,24 +11,24 @@
...
+import seedu.address.model.person.Student;
 
 /**
  * An Immutable AddressBook that is serializable to JSON format.
  */
 @JsonRootName(value = "addressbook")
 class JsonSerializableAddressBook {
 
-    public static final String MESSAGE_DUPLICATE_PERSON = "Persons list contains duplicate person(s).";
+    public static final String MESSAGE_DUPLICATE_PERSON = "Persons list contains duplicate student(s).";
```
Comment (jaysmyname) [inline src/main/java/seedu/address/storage/JsonSerializableAddressBook.java:22]: .

File: src/main/java/seedu/address/model/person/UniquePersonList.java [hunk 12]
```diff
@@ -49,30 +50,30 @@ public void add(Person toAdd) {
...
+    public void setPerson(Student target, Student editedStudent) {
+        requireAllNonNull(target, editedStudent);
 
         int index = internalList.indexOf(target);
         if (index == -1) {
             throw new PersonNotFoundException();
         }
 
-        if (!target.isSamePerson(editedPerson) && contains(editedPerson)) {
+        if (!target.isSamePerson(editedStudent) && contains(editedStudent)) {
```
Comment (jaysmyname) [inline src/main/java/seedu/address/model/person/UniquePersonList.java:65]: change

File: src/main/java/seedu/address/storage/JsonSerializableAddressBook.java [hunk 13]
```diff
@@ -37,7 +37,7 @@ public JsonSerializableAddressBook(@JsonProperty("persons") List<JsonAdaptedPers
      * @param source future changes to this will not affect the created {@code JsonSerializableAddressBook}.
      */
     public JsonSerializableAddressBook(ReadOnlyAddressBook source) {
-        persons.addAll(source.getPersonList().stream().map(JsonAdaptedPerson::new).collect(Collectors.toList()));
+        students.addAll(source.getPersonList().stream().map(JsonAdaptedStudent::new).collect(Collectors.toList()));
```
Comment (jaysmyname) [inline src/main/java/seedu/address/storage/JsonSerializableAddressBook.java:40]: .

File: src/main/java/seedu/address/storage/JsonSerializableAddressBook.java [hunk 14]
```diff
@@ -47,12 +47,12 @@ public JsonSerializableAddressBook(ReadOnlyAddressBook source) {
...
-        for (JsonAdaptedPerson jsonAdaptedPerson : persons) {
-            Person person = jsonAdaptedPerson.toModelType();
-            if (addressBook.hasPerson(person)) {
+        for (JsonAdaptedStudent jsonAdaptedStudent : students) {
+            Student student = jsonAdaptedStudent.toModelType();
+            if (addressBook.hasPerson(student)) {
                 throw new IllegalValueException(MESSAGE_DUPLICATE_PERSON);
             }
-            addressBook.addPerson(person);
+            addressBook.addPerson(student);
```
Comment (jaysmyname) [inline src/main/java/seedu/address/storage/JsonSerializableAddressBook.java:55]: .

File: src/main/java/seedu/address/ui/MainWindow.java [hunk 15]
```diff
@@ -163,8 +163,8 @@ private void handleExit() {
         primaryStage.hide();
     }
 
-    public PersonListPanel getPersonListPanel() {
-        return personListPanel;
+    public StudentListPanel getPersonListPanel() {
+        return studentListPanel;
     }
```
Comment (jaysmyname) [inline src/main/java/seedu/address/ui/MainWindow.java:168]: change method name

File: src/main/java/seedu/address/ui/StudentListPanel.java [hunk 16]
```diff
@@ -0,0 +1,49 @@
...
+import javafx.collections.ObservableList;
+import javafx.fxml.FXML;
+import javafx.scene.control.ListCell;
+import javafx.scene.control.ListView;
+import javafx.scene.layout.Region;
+import seedu.address.commons.core.LogsCenter;
+import seedu.address.model.person.Student;
+
+/**
+ * Panel containing the list of persons.
```
Comment (jaysmyname) [inline src/main/java/seedu/address/ui/StudentListPanel.java:14]: .

File: src/test/java/seedu/address/logic/LogicManagerTest.java [hunk 17]
```diff
@@ -79,18 +79,18 @@ public void execute_storageThrowsIoException_throwsCommandException() {
...
         // Execute add command
-        String addCommand = AddCommand.COMMAND_WORD + NAME_DESC_AMY + PHONE_DESC_AMY + EMAIL_DESC_AMY
-                + ADDRESS_DESC_AMY;
-        Person expectedPerson = new PersonBuilder(AMY).withTags().build();
+        String addCommand = AddCommand.COMMAND_WORD + NAME_DESC_AMY + TELEGRAM_DESC_AMY + EMAIL_DESC_AMY
+                + GITHUB_DESC_AMY;
+        Student expectedStudent = new PersonBuilder(AMY).withTutorialGroup().build();
         ModelManager expectedModel = new ModelManager();
-        expectedModel.addPerson(expectedPerson);
+        expectedModel.addPerson(expectedStudent);
```
Comment (jaysmyname) [inline src/test/java/seedu/address/logic/LogicManagerTest.java:86]: .

File: src/test/java/seedu/address/logic/commands/AddCommandIntegrationTest.java [hunk 18]
```diff
@@ -27,19 +27,19 @@ public void setUp() {
 
     @Test
     public void execute_newPerson_success() {
-        Person validPerson = new PersonBuilder().build();
+        Student validStudent = new PersonBuilder().build();
 
         Model expectedModel = new ModelManager(model.getAddressBook(), new UserPrefs());
-        expectedModel.addPerson(validPerson);
+        expectedModel.addPerson(validStudent);
```
Comment (jaysmyname) [inline src/test/java/seedu/address/logic/commands/AddCommandIntegrationTest.java:33]: .

File: src/test/java/seedu/address/logic/commands/AddCommandIntegrationTest.java [hunk 19]
```diff
@@ -27,19 +27,19 @@ public void setUp() {
...
+        assertCommandSuccess(new AddCommand(validStudent), model,
+                String.format(AddCommand.MESSAGE_SUCCESS, validStudent), expectedModel);
     }
 
     @Test
     public void execute_duplicatePerson_throwsCommandException() {
-        Person personInList = model.getAddressBook().getPersonList().get(0);
-        assertCommandFailure(new AddCommand(personInList), model, AddCommand.MESSAGE_DUPLICATE_PERSON);
+        Student studentInList = model.getAddressBook().getPersonList().get(0);
+        assertCommandFailure(new AddCommand(studentInList), model, AddCommand.MESSAGE_DUPLICATE_PERSON);
```
Comment (jaysmyname) [inline src/test/java/seedu/address/logic/commands/AddCommandIntegrationTest.java:42]: .

File: src/test/java/seedu/address/logic/commands/DeleteCommandTest.java [hunk 20]
```diff
@@ -28,20 +28,20 @@
...
-        Person personToDelete = model.getFilteredPersonList().get(INDEX_FIRST_PERSON.getZeroBased());
+        Student studentToDelete = model.getFilteredStudentList().get(INDEX_FIRST_PERSON.getZeroBased());
         DeleteCommand deleteCommand = new DeleteCommand(INDEX_FIRST_PERSON);
 
-        String expectedMessage = String.format(DeleteCommand.MESSAGE_DELETE_PERSON_SUCCESS, personToDelete);
+        String expectedMessage = String.format(DeleteCommand.MESSAGE_DELETE_PERSON_SUCCESS, studentToDelete);
 
         ModelManager expectedModel = new ModelManager(model.getAddressBook(), new UserPrefs());
-        expectedModel.deletePerson(personToDelete);
+        expectedModel.deletePerson(studentToDelete);
```
Comment (jaysmyname) [inline src/test/java/seedu/address/logic/commands/DeleteCommandTest.java:37]: .

**Review by jaysmyname — CHANGES_REQUESTED** [review]
I reviewed and found some mistakes the refactoring didn't catch. I didn't review the whole of src/test, but they should be fixed once the methods in main get renamed correctly

File: src/test/data/JsonAddressBookStorageTest/invalidStudentAddressBook.json [hunk 21]
```diff
@@ -1,8 +1,9 @@
 {
-  "persons": [ {
+  "students": [ {
     "name": "Person with invalid name field: Ha!ns Mu@ster",
-    "phone": "9482424",
+    "telegram": "@personWith",
     "email": "hans@example.com",
-    "address": "4th street"
+    "gitHub": "persons-with",
```
Comment (hqhqhq1) [inline src/test/data/JsonAddressBookStorageTest/invalidStudentAddressBook.json:6]: some small issues like line 3: "student with", and some in java docs description can change later

**Review by hqhqhq1 — COMMENTED** [review]
some small issues do not affect functionality, I am okay with merge

**Pull request merged**